react-ui/src/reducer.js

```diff
@@ -19,10 +20,23 @@ export const timerFrequency = createReducer(60, {
   [actions.setTimerFrequency]: (state, action) => action.payload
 })
 
-export const pictureFilter = createReducer(null, {
+export const pictureFilter = createReducer(initialFilterList[0], {
   [actions.setPictureFilter]: (state, action) => action.payload
 })
 
+export const pictureFilterWeight = createReducer(1.0, {
+  [actions.setPictureFilterWeight]: (state, action) => action.payload
+})
+
+export const pictureFilterList = createReducer(initialFilterList, {
+	// Put the chosen filter as first after 'none'. If filter not yet present, add it
```

Comment:
The fact this comment is needed is an indication that something is smelly.

In this case, the problem is with `setPictureFilter` - it's doing too many things at once and we need to do complicated logic in the reducer.

Instead of keeping a list of all possible filters, I'd do this instead:

```
export const customFilters = createReducer([], {
  [actions.uploadNewFilter]: (state, action) => uniqBy([action.payload].concat(state), 'label')
})
```

And handle prepending the initial filter + none in a selector in `selectors.js`.

Basically keep in redux only what should be in redux, everything non-custom out of it.